From 64c1fd6b76f145f8589ccbe2222766f147cca05f Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Tue, 13 Feb 2024 19:44:09 +0100
Subject: [PATCH] Add Managarm support

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4422b39..7a1dc40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -134,7 +134,9 @@ ecm_set_disabled_deprecation_versions(QT 5.15.2
     KF 5.101
 )
 
-add_subdirectory(greeter)
+if(PAM_REQUIRED)
+  add_subdirectory(greeter)
+endif()
 add_subdirectory(kcm)
 add_subdirectory(settings)
 
-- 
2.43.0

